src/main/java/edu/unc/lib/boxc/migration/cdm/services/ArchivalDestinationsService.java

```diff
@@ -98,6 +105,44 @@ public Map<String, String> generateCollectionNumbersToPidMapping(DestinationMapp
         return mapCollNumToPid;
     }
 
+    /**
+     * Add archival collection mapping(s) to destination mappings file
+     * @param options
+     * @throws Exception
+     */
+    public void addArchivalCollectionMappings(DestinationMappingOptions options) throws Exception {
+        Path destinationMappingsPath = project.getDestinationMappingsPath();
+
+        try (
+                BufferedWriter writer = Files.newBufferedWriter(destinationMappingsPath,
+                        StandardOpenOption.APPEND,
+                        StandardOpenOption.CREATE);
+                CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.Builder.create().build());
+        ) {
+            if (options.getFieldName() != null) {
```

Comment:
I'd suggest moving this check to the top of the method and doing either an early return or throw an error, since there isn't really a reason to open/create the file if we aren't going to write anything to it, and it'll save having to indent everything an extra time